lib/sport_ngin_aws_auditor/instance_helper.rb

```diff
@@ -109,18 +79,82 @@ def sort_through_RIs(client)
       return ris_region, ris_hash
     end
 
-    def measure_differences(instance_hash, ris_hash)
+    def measure_differences(instance_hash, ris_hash, ris_region, klass)
       differences = Hash.new()
+
+      # Setup and zone-based RI calculation
       instance_hash.keys.concat(ris_hash.keys).uniq.each do |key|
         instance_count = instance_hash.has_key?(key) ? instance_hash[key][:count] : 0
         ris_count = ris_hash.has_key?(key) ? ris_hash[key][:count] : 0
-        differences[key] = {:count => ris_count - instance_count, :region_based => false}
+        # positive count => more ris, negative count => more instances
+        differences[key] = { count: ris_count - instance_count, region_based: false }
       end
+
+      # Region-based RI calculation
+      ris_region_processor(differences, ris_region) unless ris_region.empty?
+
       differences
     end
 
+    # Assuming zone based information is already added and differences hash is initialized
+    private def ris_region_processor(differences, ris_region)
+      # Group all the same size RIs
+      # Ex: ri_group_arr = [{ instance_type: "t2.medium", platform: "Linux VPC", count: 7 }, ...]
+      ri_group_arr = group_ris_region(ris_region)
+
+      # Process each ri_group determined by instance_type and platform
+      ri_group_arr.each do |ri_group|
+        target_instance_type, target_platform, ri_count = ri_group[:instance_type], ri_group[:platform], ri_group[:count]
```

Comment:
I think this line should be separated out into three lines. Generally, it's better to not have multiple variables being set on one line (although this repo unfortunately tends to do that a lot).
```suggestion
        target_instance_type = ri_group[:instance_type]
        target_platform = ri_group[:platform]
        ri_count = ri_group[:count]
```

Reply:
Fixed